File: 0075-sort-colors/0075-sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        l, e, g = 0, 0, n-1
        
        while e <= g:
            if nums[e] == 2:
                nums[e], nums[g] = nums[g], nums[e]
                g -= 1
            elif nums[e] == 0:
                nums[e], nums[l] = nums[l], nums[e]
                l += 1
                e += 1
            else:
                e += 1

File: 0075-sort-colors/0075-sort-colors.java
class Solution {

    void swap (int [] a, int i, int j) {
        int tm = a[i];
        a[i] = a[j];
        a[j] = tm;
    }
    public void sortColors(int[] nums) {

        int l = 0, m = 0, h = nums.length-1;
        /**
         *   [2, 0, 2, 1, 1, 1]
         *    l
              m              h
             [2, 0, 2, 1, 1, 1]

            [0, 1, 2, 1, 1, 2]
             l
                m        h


         */
        while (m <= h) {
            if (nums[m] == 1){
                m++;
            } else if (nums[m] == 0) {
                swap(nums, l, m);
                l++;
                m++;
            } else {
                swap(nums, h, m);
                h--;
            }
        }
        
    }
}